### PR TITLE
adds a new module for creating show/hide or collapse/expand links

### PR DESCRIPTION
**Pull request by ushkarev**
used to show/hide a set of target elements

### CODE REVIEW

**Comment by ushkarev** [comment]
@maxf could you pls review?

File: assets/javascripts/modules/moj.show-hide.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+    cacheEls: function() {
+      this.$showHideButtons = $(this.selector);
+    },
+
+    bindEvents: function() {
+      this.$showHideButtons.on('click', this.onShowHide);
+    },
+
+    onLoad: function() {
```
Comment (maxf) [inline assets/javascripts/modules/moj.show-hide.js]: I wouldn't call it on* as it's not an event handler like onShowHide is

File: assets/javascripts/modules/moj.show-hide.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+      this.$showHideButtons.each(function() {
+        var $showHideButton = $(this);
+        if($showHideButton.is('.ShowHide-Hidden')) {
```
Comment (maxf) [inline assets/javascripts/modules/moj.show-hide.js]: .ShowHide-hidden, to follow the pattern set in _dialog.js

Comment (maxf) [inline assets/javascripts/modules/moj.show-hide.js]: add a space between 'if' and '('

File: assets/javascripts/modules/moj.show-hide.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+      var $showHideButton = $(e.target).closest(this.selector),
+        $target = $($showHideButton.attr('href'));
+      if($showHideButton.is('.ShowHide-Hidden')) {
+        $target.show();
+      } else {
+        $target.hide();
+      }
+      $showHideButton
+        .toggleClass('ShowHide-Visible')
+        .toggleClass('ShowHide-Hidden');
```
Comment (maxf) [inline assets/javascripts/modules/moj.show-hide.js]: Why 2 classes? Perhaps just visible by default?

File: assets/scss/elements/_show-hide.scss [hunk 4]
```diff
@@ -0,0 +1,32 @@
+$component: ShowHide;
+
+.#{$component} {
+  text-decoration: none;
+
+  span {
+    display: inline-block;
+    color: #000;
```
Comment (maxf) [inline assets/scss/elements/_show-hide.scss]: or $text-colour or any more generic govuk_template colour?

File: assets/scss/elements/_show-hide.scss [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  }
+  b {
```
Comment (maxf) [inline assets/scss/elements/_show-hide.scss]: better use a class. b is confusing: as it used to mean bold it's not recommended to use it without a class.

File: assets/scss/elements/_show-hide.scss [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+    margin-right: 0.5em;
+    background: #fff;
+    border-radius: 100%;
+    text-align: center;
+    vertical-align: middle;
+  }
+
+  &.ShowHide-Visible {
+    .ShowHide-Show {
+      display: none;
```
Comment (maxf) [inline assets/scss/elements/_show-hide.scss]: -Visible : display:none
confusing. See other comment above

**Pull request merged**